lib/Conversion/TritonGPUToLLVM/ControlFlowOpToLLVM.cpp

```diff
@@ -85,10 +85,21 @@ struct CallOpConversion : public ConvertOpToLLVMPattern<triton::CallOp> {
     if (!caller->hasAttr("allocation.offset")) {
       auto base = LLVM::getStackPointer(rewriter, caller);
       promotedOperands.push_back(base);
-      return promotedOperands;
+    } else {
+      auto base = LLVM::getSharedMemoryBase(loc, rewriter, targetInfo, callOp);
+      promotedOperands.push_back(base);
     }
-    promotedOperands.push_back(LLVM::getSharedMemoryBase(
-        callOp->getLoc(), rewriter, targetInfo, callOp));
+
+    auto opOffsetAttr = caller->getAttrOfType<mlir::IntegerAttr>(
+        "triton_nvidia_gpu.global_scratch_memory_offset");
```

Comment:
it's a bit weird to have this in `triton_nvidia_gpu` namespace, especially since this is in shared code